flytepropeller/pkg/controller/nodes/array/handler_test.go

```diff
@@ -379,7 +380,7 @@ func TestHandleArrayNodePhaseNone(t *testing.T) {
 			},
 			expectedArrayNodePhase:         v1alpha1.ArrayNodePhaseExecuting,
 			expectedTransitionPhase:        handler.EPhaseRunning,
-			expectedExternalResourcePhases: []idlcore.TaskExecution_Phase{idlcore.TaskExecution_QUEUED, idlcore.TaskExecution_QUEUED},
+			expectedExternalResourcePhases: []idlcore.TaskExecution_Phase{idlcore.TaskExecution_UNDEFINED, idlcore.TaskExecution_UNDEFINED},
```

Comment:
just curious, why are we changing this?

Reply:
This test case is the scenario where the ArrayNode has not yet started. We write an single node and task event for each subnode to alert admin of it's existence - mainly so the UI can display `undefined x20` to indicate that there are 20 subtasks. We previously initially reported these with the `Queued` phase, because that skips input resolution downstream allowing us to just use a mocked in-memory input value from the array rather than copy each individual subnode input to a separate blobstore location. However, we can still use `Queued` as the initial phase during downstream evaluations and use `Undefined` here to more accurately report that these tasks have not yet started.